### PR TITLE
Add new dbg.Type methods: offsetof, enum_member

### PR DESCRIPTION
**Pull request by patryk4815**

### CODE REVIEW

File: pwndbg/dbg/__init__.py [hunk 1]
```diff
@@ -765,6 +765,78 @@ def keys(self) -> List[str]:
...
+        NESTED_TYPES = (TypeCode.STRUCT, TypeCode.UNION)
+        struct_type = self
+        if nested_cyclic_types is None:
+            nested_cyclic_types = []
+
+        if struct_type.code == TypeCode.TYPEDEF:
+            struct_type = struct_type.strip_typedefs()
+
+        if struct_type.code == TypeCode.POINTER:
+            struct_type = struct_type.target().strip_typedefs()
```
Comment (patryk4815) [inline pwndbg/dbg/__init__.py]: we shouldn't resolve pointers?

File: pwndbg/dbg/__init__.py [hunk 2]
```diff
@@ -765,6 +765,78 @@ def keys(self) -> List[str]:
...
+    ) -> int | None:
+        NESTED_TYPES = (TypeCode.STRUCT, TypeCode.UNION)
+        struct_type = self
+        if nested_cyclic_types is None:
+            nested_cyclic_types = []
+
+        if struct_type.code == TypeCode.TYPEDEF:
+            struct_type = struct_type.strip_typedefs()
+
+        if struct_type.code == TypeCode.POINTER:
```
Comment (disconnect3d) [inline pwndbg/dbg/__init__.py]:
```suggestion
        elif struct_type.code == TypeCode.POINTER:
```

File: pwndbg/dbg/__init__.py [hunk 3]
```diff
@@ -765,6 +765,78 @@ def keys(self) -> List[str]:
...
+            struct_type = struct_type.target().strip_typedefs()
+
+        if struct_type.code not in NESTED_TYPES:
```
Comment (disconnect3d) [inline pwndbg/dbg/__init__.py:792]:
```suggestion
        elif struct_type.code not in NESTED_TYPES:
```

File: pwndbg/dbg/__init__.py [hunk 4]
```diff
@@ -765,6 +765,78 @@ def keys(self) -> List[str]:
...
+            return None
+
+        # TODO: improve `struct_type in nested_cyclic_types`, really slow
+        #  lldb.SBType and gdb.Type dont support Sets
+        if struct_type in nested_cyclic_types:
```
Comment (disconnect3d) [inline pwndbg/dbg/__init__.py]: should this be `elif`?

**Review by disconnect3d — APPROVED** [review]

File: pwndbg/dbg/__init__.py (hunk 3 above)
Comment (patryk4815) [reply pwndbg/dbg/__init__.py:792]: image

we are changing same variable, and it must be just `if` not `elif`

Comment (patryk4815) [reply pwndbg/dbg/__init__.py:792]: but anyway I need to think and remove this pointer type from here

File: pwndbg/dbg/__init__.py (hunk 1 above)
Comment (patryk4815) [reply pwndbg/dbg/__init__.py]: fixed

**Pull request merged**